blas64/blas64.go

```diff
@@ -0,0 +1,331 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package blas64 provides a simple interface to the float64 BLAS API.
+package blas64
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/goblas"
+)
+
+var blas64 blas.Float64 = goblas.Blas{}
+
+// Use sets the BLAS float64 implementation to be used by subsequent BLAS calls.
+// The default implementation is goblas.Blas.
+func Use(b blas.Float64) {
+	blas64 = b
+}
+
+// Implementation returns the current BLAS float64 implementation.
+//
+// Implementation allows direct calls to the current the BLAS float64 implementation
+// giving finer control of parameters.
+func Implementation() blas.Float64 {
+	return blas64
+}
+
+// Vector represents a vector with an associated element increment.
+type Vector struct {
+	Inc  int
+	Data []float64
+}
+
+// General represents a matrix using the conventional storage scheme.
+type General struct {
+	Rows, Cols int
+	Stride     int
+	Data       []float64
+}
+
+// Band represents a band matrix using the band storage scheme.
```

Comment:
I wasn't sure of this so I did a survey and it seems to be pretty much equally either way; band is shorter, and netlib and wikipedia both call it "band matrix" (with wikipedia saying "or banded".